### PR TITLE
Update version detection scheme for native

### PR DESCRIPTION
**Pull request by jerboaa**
A new version scheme got pushed in upstream GraalVM. The new version output will look similar to `java --version` output going forward. Quarkus needs to update the version detection scheme to account for that.

For GraalVM CE the version is being deduced from the `jvmci` build level (a.k.a `java.runtime.version` property). Example:
```
$ bin/java -XshowSettings:properties --version 2>&1 | grep java.runtime.version
    java.runtime.version = 20+34-jvmci-23.0-b10
```

For Mandrel, we use the vendor version which we have appropriately set for Mandrel builds. Example: `Mandrel-23.1.0-dev`

This patch works for `native-image --version` outputs of GraalVM CE:

```
$ native-image --version
native-image 20 2023-03-21
GraalVM Runtime Environment GraalVM CE (build 20+34-jvmci-23.0-b10)
Substrate VM GraalVM CE (build 20+34, serial gc)
```

and Mandrel:

```
$ native-image --version
native-image 17.0.6 2023-01-17
OpenJDK Runtime Environment Mandrel-23.1.0-dev (build 17.0.6+10)
OpenJDK 64-Bit Server VM Mandrel-23.1.0-dev (build 17.0.6+10, mixed mode)
```

Testing: 
- [x] Example `smallrye-metrics` integration test which [fails currently in CI](https://github.com/graalvm/mandrel/actions/runs/4613968709/jobs/8156830248#step:11:1278) which isn't clear, though, whether or not it's because of the version detection. Anyway, it passes with mandrel/graal ce builds with this patch.
- [x] Ran `jaxb` integration-test with `quarkus.native.debug.enabled=true` and observed that `-H:+TrackNodeSourcePosition` got added to the `native-image` invocation (that's only for `>= 23.0`). It also doesn't do the the `objcopy --strip-debug` invocation via Quarkus (as that's done upstream for `>= 23.0`).
- [x] Added test cases in `GraalVMTest.java`

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java
Comment (zakkak) [inline core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: This seems fragile as we do not know for how long we will be able to rely on the JVMCI version for figuring out the GraalVM version.

Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: Yes, agreed. We'll have to wait and see what the final version will look like. Once we know, I can update this bit.

Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: It appears the `jvmci-` bits will no longer be present in more recent GraalVM CE dev builds. As there aren't any available yet, I'll wait for them to show up before I adjust it. It seems the vendor version will give us something going forward.

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: FTR the upstream PR with the latest changes (and an ongoing discussion) is https://github.com/oracle/graal/pull/6356

Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: Here would be an alternate approach of using the `release` file (for GraalVM - if we must), fwiw: https://github.com/quarkusio/quarkus/commit/3a9502d2c3d5d20b0053b0bddb24e80034a61f1b

Comment (jerboaa) [reply core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/GraalVM.java:113]: We agreed to go with the `jvmci-` bit for now. Resolving conversation.

**Comment by zakkak** [comment]
> * Example `smallrye-metrics` integration test which [fails currently in CI](https://github.com/graalvm/mandrel/actions/runs/4613968709/jobs/8156830248#step:11:1278) which isn't clear, though, whether or not it's because of the version detection. Anyway, it passes with mandrel/graal ce builds with this patch.

This fails when passing `-H:-ParseOnce` (which happens when the version is not detected).

**Review by zakkak — APPROVED** [review]
LGTM. Thanks @jerboaa !

**Comment by jerboaa** [comment]
CI failures seem unrelated.

@gsmet Could you perhaps help get this merged? Not sure if this needs a second review, before though.

**Comment by Karm** [comment]
@jerboaa 

I am getting
```
[ERROR] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Unable to get GraalVM version from the native-image binary.
[WARNING] [io.quarkus.deployment.pkg.steps.NativeImageBuildStep] Unable to parse used Java version from native-image version string `Undefined'. Java version checks will be skipped.
```

with my latest 23.0 branch + latest JDK 20:
```
$ docker run quay.io/karmkarm/ubi-quarkus-mandrel-builder-image:23.0-java20 --version
native-image 20.0.1-internal 2023-04-18
OpenJDK Runtime Environment (build 20.0.1-internal-adhoc.karm.jdk20u)
OpenJDK 64-Bit Server VM (build 20.0.1-internal-adhoc.karm.jdk20u, mixed mode)
```

I used `--with-version-feature=20 --with-version-interim=0 --with-version-update=1` in my `configure` script....

 * Could we have the regexps amended to accommodate it? Having `internal` suffix used to be fine.
 * or could you suggest to me how do you structure the version naming for local dev builds so as it's possible to use feature, interim, update...?

**Comment by jerboaa** [comment]
@Karm Please use latest mandrel-packaging, revision `204ca7f4ee3a08ae06c8f600af20e215ecdf3f7f` and better. Be sure to set `--mandrel-version` to a parsable version of `Version.parse()` and it should end up looking like:

```
native-image 17.0.7-beta 2023-04-18
OpenJDK Runtime Environment Mandrel-23.1.0-dev (build 17.0.7-beta+5-202304062331)
OpenJDK 64-Bit Server VM Mandrel-23.1.0-dev (build 17.0.7-beta+5-202304062331, mixed mode)
```

**Comment by Karm** [comment]
Thx @jerboaa  it was indeed my fault in my dated local packaging setup. 

it's fine now:
```
$ docker run  quay.io/karmkarm/ubi-quarkus-mandrel-builder-image:23.0-java20 --version
native-image 20.0.1 2023-04-18
OpenJDK Runtime Environment Mandrel-23.0.0-dev5f40e8786574 (build 20.0.1)
OpenJDK 64-Bit Server VM Mandrel-23.0.0-dev5f40e8786574 (build 20.0.1, mixed mode)
```

**Pull request merged**
